R/prepare-admin-panel.R

```diff
@@ -181,14 +187,21 @@ prepare_admin_panel_components <- function(
         font = list(size = 16)
       )
     }
-    plot_data %>%
-      plotly::plot_ly(
-        x = ~date, y = ~value, color = ~statistic,
-        colors = c("#fbbd08", "#b21e1e", "#00827c", "#1a69a4"),
-        yaxis = ~paste0("y", id)
-      ) %>%
-      plotly::add_bars() %>%
-      plotly::subplot(nrows = n_plots, shareX = TRUE) %>%
+
+    plot_arguments <- plot_data %>%
+      dplyr::group_by(.data$id) %>%
+      dplyr::group_map(function(x, ...) {
+        x %>%
+        plotly::plot_ly(
+          x = ~date, y = ~value, color = ~statistic, colors = ~color
+        ) %>%
+          plotly::add_bars()
+      })
+
+    plot_arguments$nrows <- n_plots
+    plot_arguments$shareX <- TRUE
+
+    do.call(plotly::subplot, plot_arguments) %>%
```

Comment:
This is amazing. I had my fair share of trouble with `plotly::subplot()`. I guess the issue here is `subplot()` not behaving well with `%>%`?

Reply:
The issue was that previously it was using the `yaxis` parameter to magically separate the plots and that was no longer possible, or I don't know how to do it.

So I had to use `dplyr::group_map` to iterate on the different groups of data _(plot1, plot2, plot3 by using field `id`)_ so `subplot` could be called with `subplot(p1, p2, p3)` where p1-p3 are not variables, but elements on a list.

thus `do.call()` was needed... This is an over-complicated solution for something that should be simple.